lib/ansible/modules/database/mysql/mysql_user.py

```diff
@@ -427,14 +427,14 @@ def pick(x):
             return x
 
     for grant in grants:
-        res = re.match("GRANT (.+) ON (.+) TO '.*'@'.*'( IDENTIFIED BY PASSWORD '.+')? ?(.*)", grant[0])
+        res = re.match("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\5)? ?(.*)""", grant[0])
```

Comment:
```suggestion
        res = re.match("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\\6)? ?(.*)""", grant[0])
```

Reply:
I’ll cherry-pick the new commit as soon as it’s merged in devel

Reply:
Your PR #52278 needs more `shipit` commands to be merged